src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -113,23 +114,41 @@ public long GetMinimalEtagForReplication()
         {
             long minEtag = long.MaxValue;
 
+            DatabaseTopology topology;
             using (_server.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
-            using (ctx.OpenReadTransaction())
             {
-                var externals = _server.Cluster.ReadRawDatabaseRecord(ctx, Database.Name).ExternalReplications;
-                if (externals != null)
+                using (ctx.OpenReadTransaction())
                 {
-                    foreach (var external in externals)
+                    var dbRecord = _server.Cluster.ReadRawDatabaseRecord(ctx, Database.Name);
+                    topology = dbRecord.Topology;
+                    List <ExternalReplication> externals = dbRecord.ExternalReplications;
+                    if (externals != null)
                     {
-                        var state = GetExternalReplicationState(_server, Database.Name, external.TaskId, ctx);
-                        var myEtag = ChangeVectorUtils.GetEtagById(state.DestinationChangeVector, Database.DbBase64Id);
-                        minEtag = Math.Min(myEtag, minEtag);
+                        foreach (var external in externals)
+                        {
+                            var state = GetExternalReplicationState(_server, Database.Name, external.TaskId, ctx);
+                            var myEtag = ChangeVectorUtils.GetEtagById(state.DestinationChangeVector, Database.DbBase64Id);
+                            minEtag = Math.Min(myEtag, minEtag);
+                        }
```

Comment:
the change here is only `topology = dbRecord.Topology;` right?
if so, please revert all other changes.

Reply:
done